### ISSUE TITLE
Import extension plugin: don't suggest imports from deprecated modules

### ISSUE THREAD

**Issue by michaelpj** [haskell/haskell-language-server#2415]
### Your environment

Which OS do you use:
NixOS
Which lsp-client do you use:
emacs
Describe your project (alternative: link to the project):
cabal.project

### Steps to reproduce

With `extra` in your package db:

```
module Test where

foo :: [a] -> Int -> a
foo l i = l |? i
```

Import extension plugin will suggest importing `|?` from `Extra`, but `Extra` is deprecated (you're supposed to use `Data.List.Extra`).

### Expected behaviour

Don't suggest imports from deprecated modules (just filter them out).

### Actual behaviour

It suggests importing `|?` from `Extra`.

**Comment by pepeiborra** [comment]
Probably shouldn't suggest imports from `*Internal*` modules either.

**Comment by michaelpj** [comment]
Yeah, it should probably use the same heuristic as the completions code does.
